crates/proposer-service/src/vkey_hash.rs

```diff
@@ -0,0 +1,107 @@
+use alloy_primitives::B256;
+use serde::{Deserialize, Serialize};
+use sp1_sdk::HashableKey;
+
+pub type HashU32 = [u32; 8];
```

Comment:
Why is this hash array of u32 and not array of u8? I see that Sp1 `HashableKey` has `bytes32_raw()`, do we need to use `hash_u32()`? It would make conversions  more simple.

Reply:
> Why is this hash array of u32 and not array of u8? I see that Sp1 `HashableKey` has `bytes32_raw()`, do we need to use `hash_u32()`? It would make conversions more simple.

I'm not quite sure these are guaranteed to be the same...

Reply:
What about 
```
    /// Hash the key into a digest of bytes elements.
    fn hash_bytes(&self) -> [u8; DIGEST_SIZE * 4] {
        words_to_bytes_be(&self.hash_u32())
    }
```

Reply:
Also the idea is that eventually, this might also be used in the context of using the vkey to the proof where the `[u32; 8]` format is expected. It is also currently used in @hadjiszs's PR where the aggregated vkey hardcoded constant is defined. In general, I'd say it's better to keep closer to the "processed" representation and parse raw data at the boundaries.